pkg/plugins/runtime/k8s/webhooks/validation.go

```diff
@@ -48,72 +48,77 @@ func (h *validatingHandler) InjectDecoder(d *admission.Decoder) error {
 }
 
 func (h *validatingHandler) Handle(ctx context.Context, req admission.Request) admission.Response {
-	if req.Operation == v1beta1.Delete {
-		return admission.Allowed("")
-	}
-
 	resType := core_model.ResourceType(req.Kind.Kind)
 
-	coreRes, err := h.coreRegistry.NewObject(resType)
-	if err != nil {
-		return admission.Errored(http.StatusBadRequest, err)
-	}
-	obj, err := h.k8sRegistry.NewObject(coreRes.GetSpec())
-	if err != nil {
-		return admission.Errored(http.StatusBadRequest, err)
-	}
-	// unmarshal k8s object from the request
-	if err := h.decoder.Decode(req, obj); err != nil {
-		return admission.Errored(http.StatusBadRequest, err)
-	}
-	if resp := h.validateSync(resType, obj, req.UserInfo); !resp.Allowed {
-		return resp
-	}
-	if resp := h.validateResourceLocation(resType, obj); !resp.Allowed {
+	if resp := h.isOperationAllowed(resType, req.UserInfo, req.Operation); !resp.Allowed {
 		return resp
 	}
 
-	if err := h.converter.ToCoreResource(obj, coreRes); err != nil {
-		return admission.Errored(http.StatusInternalServerError, err)
-	}
+	switch req.Operation {
```

Comment:
Is there a benefit to using a switch with one case + default vs if-else?

Reply:
I think when you look at the code with `switch` it's more clear that there are 2 cases. Before that there was an `if` right at the beginning of the function:
```go
func (h *validatingHandler) Handle(ctx context.Context, req admission.Request) admission.Response {
	if req.Operation == v1beta1.Delete {
		return admission.Allowed("")
	}
...
```
and apparently, it wasn't noticed. No real benefit, just better readability in my opinion.